backends/bmv2/lower.cpp

```diff
@@ -153,15 +153,20 @@ struct VariableWriters {
 
 struct VariableDefinitions {
     std::map<cstring, const VariableWriters*> writers;
+    VariableDefinitions(const VariableDefinitions& other) = default;
+    VariableDefinitions() = default;
     VariableDefinitions* clone() const {
-        auto result = new VariableDefinitions();
-        result->writers = writers;
-        return result;
+        return new VariableDefinitions(*this);
     }
     VariableDefinitions* join(const VariableDefinitions* other) const {
         auto result = clone();
-        for (auto e : other->writers)
-            result->writers.emplace(e);
+        for (auto e : other->writers) {
+            auto prev = writers.find(e.first);
+            if (prev == writers.end())
+                result->writers[e.first] = e.second;
+            else
+                result->writers[e.first] = prev->second->join(e.second);
```

Comment:
This could be just

    result->writers[e.first] = result->writers[e.first]->join(e.second);